### ISSUE TITLE
Fatal Exception: NSInvalidArgumentException

### ISSUE THREAD

**Issue by newbdez33** [StreetVoice/HysteriaPlayer#98]
my users crash many times on this:

====
Fatal Exception: NSInvalidArgumentException
AVPlayerItem cannot service a seek request with a completion handler until its status is AVPlayerItemStatusReadyToPlay.

====
HysteriaPlayer.m line 293
-[HysteriaPlayer findSourceInPlayerItems:]

====

any suggestion?
Thank you very much!

**Comment by saiday** [comment]
The error log says it, do you have seeking functionality in your app?
If so, disable it until your AVPlayerItem's status update to
AVPlayerItemStatusReadyToPlay.

HysteriaPlayer had a delegate about AVPlayerItem status for you.

**Comment by newbdez33** [comment]
Hi Saiday, thanks for the reply.
But I don't have seeking function.

the seek call is from "findSourceInPlayerItems as you can see:

In HysteraPlayer.m 
```objective-c
- (BOOL)findSourceInPlayerItems:(NSUInteger)index
{
    for (AVPlayerItem *item in self.playerItems) {
        NSInteger checkIndex = [[self getHysteriaIndex:item] integerValue];
        if (checkIndex == index) {
            [item seekToTime:kCMTimeZero completionHandler:^(BOOL finished) {
                [self insertPlayerItem:item];
            }];
            return YES;
        }
    }
    return NO;
}
```

**Comment by saiday** [comment]
Sorry, my fault. I see now.

Which version of HysteriaPlayer you using? If possible, can you attach
stack log so I can trace on it.

**Comment by newbdez33** [comment]
I'm using pod version 
```pod 'HysteriaPlayer', '= 2.1.0'```

You my check the crash here:
http://crashes.to/s/dce5979ad7c

Thanks!

**Comment by saiday** [comment]
Thank you, Jack

**Comment by saiday** [comment]
@newbdez33 I've tested on `findSourceInPlayerItems` and it works well, my testing method is jump between next and previous track.

Are you jump to your current track?

A optional workaround can fix your problem, set memory cache to false. See
```
- (void)enableMemoryCached:(BOOL)memoryCache
```

**Comment by newbdez33** [comment]
Yes it's all good when I test it. and I'm not sure how to re-produce this issue anyway.
but I will try the workaround. Thank you!
